doc/sphinx/language/gallina-extensions.rst

```diff
@@ -951,83 +944,39 @@ Reserved commands inside an interactive module
     .. exn:: This is not the last opened module.
        :undocumented:
 
-.. cmd:: Module @ident := @module_expression
-
-    This command defines the module identifier :token:`ident` to be equal
-    to :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} := @module_expression
-
-       Defines a functor with parameters given by the list of :token:`module_binding` and body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} : @module_type := @module_expression
-
-       Defines a functor with parameters given by the list of :token:`module_binding` (possibly none), and output module type :token:`module_type`,
-       with body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type} := @module_expression
-
-       Defines a functor with parameters given by module_bindings (possibly none) with body :token:`module_expression`.
-       The body is checked against each :n:`@module_type__i`.
-
-    .. cmdv:: Module @ident {* @module_binding} := {+<+ @module_expression}
-
-       is equivalent to an interactive module where each :token:`module_expression` is included.
-
-.. cmd:: Module Type @ident
-
-   This command is used to start an interactive module type :token:`ident`.
-
-   .. cmdv:: Module Type @ident {* @module_binding}
-
-      Starts an interactive functor type with parameters given by :token:`module_bindings`.
-
 
 Reserved commands inside an interactive module type:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. cmd:: Include @module
-
-   Same as ``Include`` inside a module.
-
-.. cmd:: Include {+<+ @module}
-
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
-
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
-
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
```

Comment:
The special meaning of `Parameter` *et al.* in interactive module types should be explained in the section on "Commands inside interactive modules and module types" even if the command is not shown here because it was already defined in the Gallina chapter. In particular, the meaning of the `Inline` annotation which was not explained in the previous chapter should still be explained here. In the previous chapter, there should be a mention that these commands take a (slightly) different meaning when inside interactive module types, and in particular that the `Inline` annotation is relevant only in this case.

Reply:
Sounds terrific, but I have little idea what should be written on this here or elsewhere.  You mean to add a new section "Commands inside ..."?  I suppose the paragraph below the `Parameter` description containing "Outside of a section, these are equivalent to :n:`Local Parameter`." should say "section or module"?

By "`Inline` annotation", you mean the `Inline` clause in Parameter et. al.?  IMHO "clause" is better because it more clearly implies syntax, whereas "annotation" and some similar words sound like they may have specific definitions within Coq.  No idea what `Inline` does here.

If you give me the facts in some detail (outline is fine), I will come up with the language.  Which ever way you prefer.

Reply:
> You mean to add a new section "Commands inside ..."?

I meant that the section "Reserved commands inside..." could simply be renamed "Commands inside..." to include commands that are not reserved but change meaning inside interactive modules / module types.

> I suppose the paragraph below the `Parameter` description containing "Outside of a section, these are equivalent to :n:`Local Parameter`." should say "section or module"?

No, I don't think that needs changing.

>  IMHO "clause" is better because it more clearly implies syntax

Yes.

> No idea what `Inline` does here.

Well, it is documented here, even though I cannot yet decipher what this means myself. For sure, we'll need to improve this once we start really working on module documentation (in particular with the merging of the module chapter and this section).

Reply:
Then if you're good with this part as it is, please resolve.  Or elaborate.

Reply:
The only thing remaining is the `Inline` documentation which is also tracked in other comments below.

Reply:
Think I got it.